fix some fc-hba are missing the statistics/*

chijiajian opened the pull request:
Some FC-HBA cards are missing the /sys/class/fc_host//statistics/* directory and files, for example, the Huawei Technologies Co., Ltd. Hi1822 Family (2*16G FC) (rev 45) card with ID 3e:00.0. However, there are corresponding port_name, port_state, etc., under /sys/class/fc_host//. It is recommended to add a check in the parseFibreChannelStatistics function in the class_fibrechannel.

ZStack-Robot committed:
fix some fc-hba are missing the statistics/*

Signed-off-by: jiajian.chi <jiajian.chi@zstack.io>

ZStack-Robot committed:
Signed-off-by: jiajian.chi <jiajian.chi@zstack.io>

 On branch fixFibrechannel
 Changes to be committed:
	modified:   sysfs/class_fibrechannel.go

Signed-off-by: jiajian.chi <jiajian.chi@zstack.io>

SuperQ commented on sysfs/class_fibrechannel.go:
```diff
@@ -153,6 +153,10 @@ func parseFibreChannelStatistics(hostPath string) (*FibreChannelCounters, error)
 
 	path := filepath.Join(hostPath, "statistics")
+	if _, err := os.Stat(path); os.IsNotExist(err) {
+		fmt.Printf("Directory %s does not exist, skipping.\n", path)
```
Please do not put `Printf` statements in this library.

chijiajian replied:
I have removed the corresponding Printf statements. Please review again. Thank you!

chijiajian committed:
Merge branch 'prometheus:master' into fixFibrechannel

chijiajian committed:
Update class_fibrechannel.go

delete Printf

Signed-off-by: chijiajian <33536362+chijiajian@users.noreply.github.com>

discordianfish commented on sysfs/class_fibrechannel.go:
```diff
@@ -153,6 +153,9 @@ func parseFibreChannelStatistics(hostPath string) (*FibreChannelCounters, error)
+		return &counters, nil
+	}
 	files, err := os.ReadDir(path)
 	if err != nil {
```
Why not check if the error is IsNotExist?

chijiajian replied:
I think it's better to just check if the statistics directory exists, instead of worrying about other errors like permission or filesystem issues. If the directory isn't there, the  function can keep going instead of stopping for fibrechannel statistics not found

chijiajian committed:
Merge branch 'master' into fixFibrechannel

chijiajian committed:
Improve error handling in parseFibreChannelStatistics

Refactor error handling for statistics directory check.

Signed-off-by: chijiajian <33536362+chijiajian@users.noreply.github.com>

chijiajian reviewed (commented):
Fixed — now explicitly checking for fs.ErrNotExist and returning other errors.

chijiajian committed:
Merge branch 'prometheus:master' into fixFibrechannel

chijiajian committed:
Fix error check for non-existent statistics directory

Signed-off-by: chijiajian <33536362+chijiajian@users.noreply.github.com>
Signed-off-by: jiajian.chi <jiajian.chi@zstack.io>

chijiajian reviewed (commented):
fix: undefined: fs

chijiajian reviewed (commented):
Updated to check errors.Is(err, os.ErrNotExist). Thanks!

chijiajian committed:
Merge branch 'prometheus:master' into fixFibrechannel

chijiajian committed:
Remove redundant os.Stat call, use os.ReadDir error directly

ReadDir already returns os.ErrNotExist when the directory does not
exist, so no need to run Stat before ReadDir.

Signed-off-by: chijiajian <33536362+chijiajian@users.noreply.github.com>

chijiajian committed:
Merge branch 'master' into fixFibrechannel

chijiajian commented:
Hi @SuperQ @discordianfish, I addressed the last review comment in a3c8569 by removing the redundant os.Stat call and handling os.ErrNotExist directly from os.ReadDir.

  The review threads are now resolved/outdated. Could you please re-run CI for the latest commit and take another look when you have a chance? Thanks!

SuperQ commented on sysfs/class_fibrechannel.go:
```diff
@@ -154,6 +154,10 @@ func parseFibreChannelStatistics(hostPath string) (*FibreChannelCounters, error)
 	files, err := os.ReadDir(path)
 	if err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			// statistics directory not present, skip
```
```suggestion
```

SuperQ committed:
Apply suggestions from code review

Co-authored-by: Ben Kochie <superq@gmail.com>
Signed-off-by: Ben Kochie <superq@gmail.com>
